updatecli/updatecli.d/windows-2022-images.yml

```
@@ -0,0 +1,42 @@
+---
+name: Bump azure `windows 2022` image version
+
+scms:
+  default:
+    kind: github
+    spec:
+      user: "{{ .github.user }}"
+      email: "{{ .github.email }}"
+      owner: "{{ .github.owner }}"
+      repository: "{{ .github.repository }}"
+      token: "{{ requiredEnv .github.token }}"
+      username: "{{ .github.username }}"
+      branch: "{{ .github.branch }}"
+
+sources:
+  lastReleaseVersion:
+    kind: shell
+    name: Get the latest `windows 2022` image version
+    spec:
+      command: az login --service-principal -u $AZURE_CLIENT_ID -p $AZURE_CLIENT_SECRET -t $AZURE_TENANT_ID && az vm image list --location eastus --publisher MicrosoftWindowsServer --offer WindowsServer --sku 2022-datacenter-core-g2 --all --query "[?offer=='WindowsServer'].version" -o tsv | sort -u | tail -n 1
```

```suggestion
      # Reuse environment variables from parent shell
      environments:
        - name: PATH # If `az` is in a non-usual installation path
        - name: AZURE_CLIENT_ID
        - name: AZURE_CLIENT_SECRET
        - name: AZURE_TENANT_ID
      command: az login --service-principal -u $AZURE_CLIENT_ID -p $AZURE_CLIENT_SECRET -t $AZURE_TENANT_ID && az vm image list --location eastus --publisher MicrosoftWindowsServer --offer WindowsServer --sku 2022-datacenter-core-g2 --all --query "[?offer=='WindowsServer'].version" -o tsv | sort -u | tail -n 1
```

Otherwise the command fails with the error `ERROR: argument --username/-u: expected one argument` because the environment variable is empty. Updatecli needs to specify explicitly the variables to pass to the child shell (to avoid passing sensitive values accidentally): https://www.updatecli.io/docs/plugins/resource/shell/#_environments